File: pset1/greedy.c
#include <stdio.h>
#include <cs50.h>


int main(void)
{
    float change;
    int counter = 0;
    do
    {
        printf("How much change is owed?\n");
        change = get_float();
        if (change < 0)
        {
            printf("A negative amount of change cannot be given! Please enter a positive value:\n");
        }
        
    }
    while (change < 0);
    
    int cents = 100 * change;
    //printf("%i\n", cents);
    
    while(cents >= 25)
    {
        counter = counter + 1;
        //cents = cents - 25;
        cents -= 25;
    }
    
    //printf("%i , %i\n", counter, cents);
    
    while(cents >= 10)
    {
        counter = counter + 1;
        cents = cents - 10;
    }
    
    while(cents >= 5)
    {
        counter = counter + 1;
        cents = cents - 5;
    }
    
    while(cents >= 1)
    {
        counter = counter + 1;
        cents = cents - 1;
    }
    
    printf("Total amount of coins used: %i\n", counter);
    
}

//in the future, try program that prints all different ways to distribute change

File: pset6/greedy.py
import cs50

def main():
    
    while True:
        print("How much change is needed? ", end="")
        change = cs50.get_float()
        
        if change < 0:
            print("A negative amount of change cannot be given! Please enter a positive value")
            
        
        if change >= 0:
            break
        
    cents = 100 * change   
    CalculateChange(cents)

def CalculateChange(cents):
    
    counter = 0 
    
    while cents >= 25:
        counter += 1
        cents -= 25
        
    while cents >= 10:
        counter += 1 
        cents -= 10
        
    while cents >= 5:
        counter += 1 
        cents -= 5 
        
    while cents >= 1:
        counter += 1 
        cents -= 1
        
    print("Total amount of coins used: {}".format(counter))
    
if __name__ == "__main__":
    main()